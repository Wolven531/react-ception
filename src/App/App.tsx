import React, { useEffect } from 'react'
import './App.css'

export function App() {
	useEffect(() => {
		console.log('[App] loaded, about to require App2')

		require('../App2')
	}, [])

	const handleClick = () => {
		console.log('[App] button was clicked')
	}

	return (
		<div
			className="app"
			id="app"
		>
			<h1>React 18</h1>
			<p>App 1</p>
			<button onClick={handleClick}>App button</button>
			<hr />
			<div id="placeholder"></div>
		</div>
	)
}
