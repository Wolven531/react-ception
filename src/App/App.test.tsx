import React from 'react'
import { render } from '@testing-library/react'
import { App } from './App'

describe('App', () => {
	it('renders view that matches snapshot', () => {
		const view = render(<App />)

		expect(view.asFragment()).toMatchSnapshot()
	})
})
