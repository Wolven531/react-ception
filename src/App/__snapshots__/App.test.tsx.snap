// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders view that matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="app"
    id="app"
  >
    <h1>
      React 18
    </h1>
    <p>
      App 1
    </p>
    <button>
      App button
    </button>
    <hr />
    <div
      id="placeholder"
    >
      <div
        class="app2"
      >
        <p>
          App 2 (injected)
        </p>
        <button>
          App 2 button
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
